elyra/pipeline/processor_local.py

```diff
@@ -112,17 +106,77 @@ def _sort_operation_dependencies(operations_by_id: dict, ordered_operations: lis
                                                                         parent_operation)
             ordered_operations.append(operation)
 
-    def _get_envs(self, operation: Operation) -> Dict:
-        """Operation stores environment variables in a list of name=value pairs, while
-           subprocess.run() requires a dictionary - so we must convert.  If no envs are
-           configured on the Operation, the existing env is returned, otherwise envs
-           configured on the Operation are overlayed on the existing env.
-        """
-        envs = os.environ.copy()
-        for nv in operation.env_vars:
-            nv_pair = nv.split("=")
-            if len(nv_pair) == 2:
-                envs[nv_pair[0]] = nv_pair[1]
-            else:
-                self.log.warning(f"Could not process environment variable entry `{nv}`, skipping...")
-        return envs
+
+class OperationProcessor(ABC):
+
+    def __init__(self):
+        self.log = log.get_logger()
+
+    @abstractmethod
+    def operation_name(self) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def process(self, operation: Operation, filepath: str):
+        raise NotImplementedError
+
+
+class FileOperationProcessor(OperationProcessor):
+    _operation_name: str
+
+    def __init__(self, root_dir: str):
+        super(FileOperationProcessor, self).__init__()
+        self._root_dir = root_dir
+
+    @property
+    def operation_name(self) -> str:
+        return self._operation_name
+
+    async def process(self, operation: Operation):
+        filepath = get_absolute_path(self._root_dir, operation.filename)
+        if not os.path.exists(filepath):
+            raise FileNotFoundError(f'Could not find {filepath}')
+        if not os.path.isfile(filepath):
+            raise ValueError(f'Not a file: {filepath}')
+
+        file_dir = os.path.dirname(filepath)
+        file_name = os.path.basename(filepath)
+
+        self.log.debug(f'Processing: {filepath}')
+
+        argv = self._create_execute_command(filepath, file_dir)
+        envs = operation.env_vars_as_dict
+        t0 = time.time()
+        try:
+            subprocess.run(argv, env=envs, check=True)
+        except Exception as ex:
+            self.log.error(f'Internal error executing {filepath}')
+            raise RuntimeError(f'Internal error executing {filepath}') from ex
+
+        t1 = time.time()
+        duration = (t1 - t0)
+        self.log.debug(f'Execution of {file_name} took {duration:.3f} secs.')
+
+    @abstractmethod
+    def _create_execute_command(self, filepath: str, cdw: str) -> list:
+        pass
```

Comment:
This should probably raise `NotImplementedError`, otherwise, it's saying that a subclass is not really required to override this method - which will result in a failure when `subprocess.run()` is invoked.